core/src/main/java/org/elasticsearch/common/geo/builders/ShapeBuilder.java

```diff
@@ -728,7 +729,17 @@ public static ShapeBuilder parse(XContentParser parser, GeoShapeFieldMapper shap
             Distance radius = null;
             CoordinateNode node = null;
             GeometryCollectionBuilder geometryCollections = null;
-            Orientation requestedOrientation = (shapeMapper == null) ? Orientation.RIGHT : shapeMapper.fieldType().orientation();
+
+            Map<String, Object> parseOptions = new HashMap<>();
```

Comment:
Why do we need a map to store two options? can we not just store the options in two local variables here. If/when this expands into more parse options and its too big to pass around as separate variables we could create a class to hold these options.